### ISSUE TITLE
Full Random Beacon DKG integration

### ISSUE THREAD

**Issue by pdyraga** [keep-network/keep-core#3044]
Depends on #3043 
Full integration with `RandomBeacon` DKG: `submitDkgResult`, `challengeDkgResult`, and `approveDkgResult`.

**Comment by mswilkison** [comment]
## Repurposing this as the tracking epic for the real random beacon

Consolidating #3043, #3045, #3046, and #3098 here. They are not independent tasks — all four are blocked on the same missing layer (wiring `pkg/chain/ethereum/beacon.go` to the already-generated `RandomBeacon` bindings), and that layer is deliberately deferred, not forgotten. This comment records why, so the dependency is visible in the tracker instead of implied by four ambiguous tickets.

### Current state (verified on `main`)

The beacon is split in two, and only one half is dormant:

- **Sortition / staking half — live.** `beacon.Initialize` is called unconditionally at `cmd/start.go:145` and runs `sortition.MonitorPool` with a `BetaOperatorPolicy`. Every method behind it is a real on-chain call (`JoinSortitionPool` `pkg/chain/ethereum/beacon.go:195`, `IsOperatorInPool` `:179`, `UpdateOperatorStatus` `:202`, `IsEligibleForRewards` `:209`, `IsChaosnetActive` `:226`, `IsBetaOperator` `:232`). `BeaconSortitionPool` is deployed on mainnet (`0x4e4cBA3779d56386ED43631b4dCD6d8EacEcBCF6`). **This half must not be touched by any cleanup.**
- **Protocol half — stubbed.** DKG, relay, and group methods return `errNotImplemented` / empty subscriptions, since `91b0092c9` (2022-12-02, "Disabled mock random beacon functionality"). At runtime the client subscribes to DKG/relay events that can never fire; it does not error.

### What produces randomness today

The sole consumer is `WalletRegistry.sol:821` (`randomBeacon.requestRelayEntry(this)`), used to select a wallet signing group. `randomBeacon` is a mutable pointer, and the deploy flow repoints it to the stub:

```
solidity/ecdsa/deploy/04_upgrade_random_beacon_chaosnet.ts:15-20
  execute("WalletRegistry", ..., "upgradeRandomBeacon", RandomBeaconChaosnet.address)
```

gated on the `useRandomBeaconChaosnet` tag, which mainnet, sepolia, and development all carry (`solidity/ecdsa/hardhat.config.ts:150,158,166`). So on mainnet the active source is `RandomBeaconChaosnet` (`0xF23fc18EcE6Afe7fb57ba59979e8600bb4B0294a`) — a deterministic `keccak` chain seeded to Euler's number — while the real `RandomBeacon` (`0x5499f54b4A1CB4816eefCf78962040461be3D80b`) sits deployed and unused. The deploy script's own comment names the blocker: *"a temporary solution to enable usage of `WalletRegistry` before the random beacon functionalities in the client are ready."* Those functionalities are this epic.

### Why it is deferred (not abandoned)

The stub's only weakness is that group selection is predictable. That matters when an adversary can be admitted as several operators and grind group composition to cross a signing threshold. Under the current **permissioned beta-operator set**, admission control — not randomness quality — is the real defense, so the stub is fit for the current trust model.

The real beacon becomes a hard requirement only if participation goes permissionless. Per the project roadmap, permissionless would only ever come **after FROST/ROAST**. That yields the dependency chain:

**FROST/ROAST GA → permissionless-participation decision → (if yes) finish the real beacon.**

It is therefore off the critical path for at least the current and next phases.

### What finishing it requires

Bindings already exist; only wiring is missing — `SubmitDkgResult`, `ChallengeDkgResult`, `ApproveDkgResult`, `SubmitRelayEntry(entry, groupMembers)`, `SubmitRelayEntry0(entry)` in the generated `RandomBeacon` bindings are never called. Two extra gaps: the Go interface (`pkg/beacon/chain/chain.go:88-111`) does not model challenge/approve, and `CalculateDKGResultHash` (`beacon.go:332`) hashes only `GroupPublicKey ++ Misbehaved` with no operators-list hash.

### Disposition of the consolidated issues

- **#3043** react to group-creation requests — folded here.
- **#3045** react to relay requests — folded here.
- **#3046** submit relay entry — folded here.
- **#3098** group-operators unit tests — part (a) (misbehavior list forming) is already **done** via `TestResolveGroupOperators` (`pkg/beacon/dkg/dkg_test.go:163`). Part (b) (off-chain vs on-chain operators-list hash) is folded here — it is unwritable until the operators-list hash exists, which is part of this epic.

**Blocked on:** FROST/ROAST GA, then a permissionless-participation decision. Not scheduled until both resolve.
